### DISCUSSION TITLE
How to set titlebar context menu to dark theme?

### DISCUSSION THREAD

**Discussion by KiyanYang** [microsoft/WindowsAppSDK#2967] [Q&A] [answered]
I have customized the titlebar, but I don't know how to set the dark theme.

<img width="364" alt="Snipaste_2022-09-14_17-39-47" src="https://user-images.githubusercontent.com/50903774/190120213-e351b7e4-f83b-4cfd-bdd5-ee26b74b72d1.png">

How to set titlebar context menu to dark theme, Like "Settings" App (Windows 11 22H2 22622.436)

<img width="498" alt="Snipaste_2022-09-14_17-35-58" src="https://user-images.githubusercontent.com/50903774/190119359-6be91e1e-5aaf-48b1-8294-a39f36860b46.png">

**Comment by castorix** [comment, marked as answer]
It works for me with _SetPreferredAppMode_, but the OP in this thread said it did not work for him : [#2943](https://github.com/microsoft/WindowsAppSDK/issues/2943)

**Reply by DarranRowe** [reply]
On Windows 11, SetPreferredAppMode isn't exported by name from uxtheme.dll. That is the nature of undocumented functions after all.

**Reply by castorix** [reply]
> On Windows 11, SetPreferredAppMode isn't exported by name from uxtheme.dll. That is the nature of undocumented functions after all.

On Windows 10 also (ordinal 135)

**Reply by DarranRowe** [reply]
Well, there is still a major problem then.
Explorer, which does have a dark system menu even on the initial Windows 11 release, doesn't touch ordinal 135 in uxtheme.dll.
This means two possible things.
1) The function changed to a different ordinal.
2) The function was just removed and Explorer uses another method of changing the system menu on Windows 11. It does use a few of the menu related functions.

**Reply by KiyanYang** [reply]
It works for me. And it responds to the system settings, which is very good (Windows 11 22H2 22622.436).

<img width="365" alt="Snipaste_2022-09-15_10-29-48" src="https://user-images.githubusercontent.com/50903774/190302465-6d7ebea1-4950-4438-bf8e-0504a09d3673.png">

```cs
protected override void OnLaunched(Microsoft.UI.Xaml.LaunchActivatedEventArgs args)
{
    m_window = new MainWindow();
    m_window.Activate();
    SetPreferredAppMode(PreferredAppMode.AllowDark);
}

[DllImport("uxtheme.dll", EntryPoint = "#135", SetLastError = true, CharSet = CharSet.Unicode)]
private static extern int SetPreferredAppMode(PreferredAppMode preferredAppMode);

enum PreferredAppMode
{
    Default,
    AllowDark,
    ForceDark,
    ForceLight,
    Max
};
```

In addition, set the DWMWA_USE_IMMERSIVE_DARK_MODE attribute to TRUE which can set the titlebar to dark.
Thanks @castorix @DarranRowe.

**Comment by DarranRowe** [comment]
There is a couple of important things that was not mentioned here.
The settings you are showing is obviously the Windows 11 settings window.

![Screenshot 2022-09-14 185450](https://user-images.githubusercontent.com/52577874/190231001-7c5901d4-e740-4fa4-a3e5-d3ae81bfb588.png)

However, the initial Windows 11 release doesn't show that context menu in dark mode. What version of Windows 11 are you using?
What's more, you are showing the customised title bar, but not what it looks like un-customised. Have you set the DWMWA_USE_IMMERSIVE_DARK_MODE attribute to TRUE when you detect dark mode? I'll do a quick test to see how it goes myself, but it will take a little while since I have to update one of my devices to 22H1.

**Reply by KiyanYang** [reply]
I'm using Windows 11 22H2 (22622.436). I didn't know about DWMWA_USE_IMMERSIVE_DARK_MODE  before. Thanks for reminding me, I'll try it.

**Reply by KiyanYang** [reply]
I created a new WinUI app and set the DWMWA_USE_IMMERSIVE_DARK_MODE  attribute to true. But only the titlebar into dark, the context menu is still light.

<img width="357" alt="Snipaste_2022-09-15_10-13-33" src="https://user-images.githubusercontent.com/50903774/190297228-fea783de-4ea2-4df3-90d2-c7ecdd7eb1d6.png">
